### ISSUE TITLE
Deploy, verify deployment and revert immediately (even if it does not fail)

### ISSUE THREAD

**Issue by dlawrences** [sqitchers/sqitch#569]
Hey team

Is there any specific argument or command that one can use to achieve the following steps without writing additional shell logic?

1. Deploy changes from a sqitch plan to a target
2. Verify them
3. Always revert to the initial starting change before deployment (which may not be @ROOT)

I know one could do:

`sqitch deploy <target> --verify`
`<shell cmd to determine the first change to rollback to>`
`sqitch revert --to <change determined above>`

But I am trying to circumvent writing the additional shell logic to parse the deploy/plan output if not 100% required. I am trying to bake this into CI logic for Snowflake Pull Requests.

Thank you,
Laurentiu

**Comment by theory** [comment]
No, but you can use [`sqitch log`]() to capture the currently-deployed change before you run `sqitch deploy`, something like:

```
change=$(sqitch log --event deploy -f format:%H -n 1)
sqitch deploy <target> --verify
if [-z "$change" ]; then
    sqitch revert -y <target>
else
    sqitch revert -y <target> --to $change
fi
```

**Comment by dlawrences** [comment]
Thanks @theory, very helpful!

**Comment by dlawrences** [comment]
I wanted to just add my resolution on this one: I had to move away from using `sqitch log --event deploy` as this reads from `SQITCH.EVENTS` records with `EVENT` = "deploy" which is obviously not going to always point to the state of the environment to which we would want to auto-revert (the latest deploy event may not point to the actual latest changeset active on the environment as that may have already been reverted).

I've been able to identify the environment state prior to deployment by doing:

```
change=$(sqitch status --target <target> | sed -n -e 's/^.*Change: //p' | awk '{$1=$1;print}')
```

Which is basically extracting the latest changeset hash active on the environment prior to deployment. And from then onwards, it's the same you added above @theory, thanks for the guidance!

**Comment by theory** [comment]
>the latest deploy event may not point to the actual latest changeset active on the environment as that may have already been reverted

🤦🏻  Oh, right.

Would be nice to have an option on `--status` to just emit the name or sha of the last deployed change. I'm gonna make this ticket a feature request for that.

**Comment by dlawrences** [comment]
Great stuff!
